### PR TITLE
feat: Add data product `wholesale_settlement_reports` amount views

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description

Added views for accessing amounts results (formerly known as wholesale results) for settlement reports

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202408051301_create_settlement_report_monthly_amounts_per_charge_v1.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+CREATE VIEW IF NOT EXISTS {CATALOG_NAME}.{WHOLESALE_SETTLEMENT_REPORTS_DATABASE_NAME}.monthly_amounts_per_charge_v1 as
+SELECT calculation_id,
+       calculation_type,
+       calculation_version,
+       result_id,
+       grid_area_code,
+       energy_supplier_id,
+       time,
+       "P1M" as resolution,
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/datamigration/migration_scripts/202408051301_create_settlement_report_monthly_amounts_per_charge_v1.sql:9]: Is this needed in a "monthly" data set?

File: source/databricks/calculation_engine/tests/contracts/data_products_databases_and_schemas.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ def get_data_product_databases(spark: SparkSession) -> List[Database]:
         "wholesale_results_internal",
         "wholesale_internal",
         "wholesale_settlement_reports",
+        "wholesale_results",
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/contracts/data_products_databases_and_schemas.py:37]: Isn't this a data product? (And settlement reports as well)

Comment (JesperGomes) [reply source/databricks/calculation_engine/tests/contracts/data_products_databases_and_schemas.py:37]: Oops. Thanks

File: source/databricks/calculation_engine/package/datamigration/migration_scripts/202408051301_create_settlement_report_monthly_amounts_per_charge_v1.sql (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202408051301_create_settlement_report_monthly_amounts_per_charge_v1.sql:9]: Not really. I can ask Raccoons if they are okay with us removing it

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/datamigration/migration_scripts/202408051301_create_settlement_report_monthly_amounts_per_charge_v1.sql]: I have agreed with Raccoons that we can remove it. Also, they suggested that we remove currency - so I have done that also

File: source/databricks/calculation_engine/contracts/data_products/wholesale_settlement_reports/amounts_per_charge_v1.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import pyspark.sql.types as t
+
+nullable = True
+
+amounts_per_charge_v1 = t.StructType(
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/databricks/calculation_engine/contracts/data_products/wholesale_settlement_reports/amounts_per_charge_v1.py:5]: We have a task on the story to update the documentation

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**